### PR TITLE
Skills specs

### PR DESCRIPTION
**Pull request by DeeDee88**
# Description
- add specs to skills feature

### Before merge checklist
- [ ] CircleCI is passing
- [ ] Proper labels are set

### CODE REVIEW

File: spec/features/user_skill_rates_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  let!(:developer) { create(:user, :developer, skills: skills_range + skills_boolean) }
+
+  before do
+    log_in_as developer
+    user_skill_rates_page.load
+  end
+
+  def skill_rate_trigger(num, event_name)
+    page.execute_script "$('.glyphicon.skill__rate[data-rate=\"#{num}\"]:first').trigger('#{event_name}')"
```
Comment (netguru-hound) [inline spec/features/user_skill_rates_spec.rb]: Line is too long. [106/100]

File: spec/features/user_skill_rates_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+require 'spec_helper'
+
+describe 'User skill rates page', js: true do
+  let(:user_skill_rates_page) { App.new.user_skill_rates_page }
+
+  let(:skill_category_1) { create :skill_category, name: 'backend' }
+  let(:skill_category_2) { create :skill_category, name: 'ios' }
+
+  let(:skills_range) { create_list :skill, 2, :with_range_rate_type, skill_category: skill_category_1 }
+  let(:skills_boolean) { create_list :skill, 2, :with_boolean_rate_type, skill_category: skill_category_2 }
```
Comment (netguru-hound) [inline spec/features/user_skill_rates_spec.rb]: Line is too long. [107/100]

File: spec/features/user_skill_rates_spec.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
+require 'spec_helper'
+
+describe 'User skill rates page', js: true do
+  let(:user_skill_rates_page) { App.new.user_skill_rates_page }
+
+  let(:skill_category_1) { create :skill_category, name: 'backend' }
+  let(:skill_category_2) { create :skill_category, name: 'ios' }
+
+  let(:skills_range) { create_list :skill, 2, :with_range_rate_type, skill_category: skill_category_1 }
```
Comment (netguru-hound) [inline spec/features/user_skill_rates_spec.rb]: Line is too long. [103/100]

File: spec/factories/skill_category.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 FactoryGirl.define do
   factory :skill_category do
-    name { Faker::Name.name }
+    name { Faker::Color.color_name.gsub(' ', '_') }
```
Comment (netguru-hound) [inline spec/factories/skill_category.rb]: Use tr instead of gsub.

File: spec/features/projects_spec.rb [hunk 5]
```diff
@@ -131,9 +136,11 @@
   end
 
   describe 'managing people in projects' do
-    before { projects_page.project_types.active_tab.click }
 
     describe 'adding member to project' do
+      before { go_to_active_tab }
+
+
```
Comment (netguru-hound) [inline spec/features/projects_spec.rb]: Extra blank line detected.

File: spec/features/user_skill_rates_spec.rb [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+
+  context 'when user does not have any marked skills' do
+    before do
+      log_in_as developer
+      user_skill_rates_page.load
+    end
+
+    def skill_rate_trigger(num, event_name)
+      page.execute_script "$('.glyphicon.skill__rate[data-rate=\"#{num}\"]:first').trigger('#{event_name}')"
```
Comment (netguru-hound) [inline spec/features/user_skill_rates_spec.rb]: Line is too long. [108/100]

File: spec/features/user_skill_rates_spec.rb [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  let(:skill_category_1) { create :skill_category, name: 'backend' }
+  let(:skill_category_2) { create :skill_category, name: 'ios' }
+
+  let(:skills_range) { create_list :skill, 2, :with_range_rate_type, skill_category: skill_category_1 }
+  let(:skills_boolean) { create_list :skill, 2, :with_boolean_rate_type, skill_category: skill_category_2 }
+
+  let!(:developer) { create(:user, :developer, skills: skills_range + skills_boolean) }
+
+
+
```
Comment (netguru-hound) [inline spec/features/user_skill_rates_spec.rb]: Extra blank line detected.

File: spec/features/user_skill_rates_spec.rb [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+  let(:skill_category_1) { create :skill_category, name: 'backend' }
+  let(:skill_category_2) { create :skill_category, name: 'ios' }
+
+  let(:skills_range) { create_list :skill, 2, :with_range_rate_type, skill_category: skill_category_1 }
+  let(:skills_boolean) { create_list :skill, 2, :with_boolean_rate_type, skill_category: skill_category_2 }
+
+  let!(:developer) { create(:user, :developer, skills: skills_range + skills_boolean) }
+
+
```
Comment (netguru-hound) [inline spec/features/user_skill_rates_spec.rb]: Extra blank line detected.

File: spec/features/user_skill_rates_spec.rb [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    end
+
+    it 'selects one star in the boolean rate' do
+      star = user_skill_rates_page.skill_rate1.first
+      expect(star[:class]).to_not include('selected')
+      skill_rate_trigger(1, 'click')
+      expect(star[:class]).to include('selected')
+    end
+
+    it 'selects two stars in the boolean rate' do
```
Comment (DariuszMusielak) [inline spec/features/user_skill_rates_spec.rb]: Skill with boolean rate has only one rate start.

File: spec/features/user_skill_rates_spec.rb [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+      star = user_skill_rates_page.skill_rate2.first
+      expect(star[:class]).to_not include('selected')
+      skill_rate_trigger(2, 'click')
+      expect(star[:class]).to include('selected')
+    end
+
+    it 'selects three stars in the boolean rate' do
```
Comment (DariuszMusielak) [inline spec/features/user_skill_rates_spec.rb]: same here

File: spec/features/user_skill_rates_spec.rb [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+      expect do
+        user_skill_rates_page.skill_note.first.set 'test note'
+        user_skill_rates_page.wait_for_save_alert
+      end.to change { skill_rate.reload.note }.from('').to('test note')
+    end
+  end
+
+  context 'when user have marked skills' do
+    before do
+      developer.user_skill_rates.update_all(rate: 2)
```
Comment (DariuszMusielak) [inline spec/features/user_skill_rates_spec.rb]: This is not trou for skills which has rate scale: boolean. Max rate value is equal to 1.

**Pull request merged**